crates/ide-completion/src/render.rs

```diff
@@ -334,21 +336,45 @@ fn render_resolution_path(
             }
         }
     }
-    if let ScopeDef::Local(local) = resolution {
-        let ty = local.ty(db);
+
+    let mut set_item_relevance = |is_local: bool, ty: Type| {
         if !ty.is_unknown() {
             item.detail(ty.display(db).to_string());
         }
 
         item.set_relevance(CompletionRelevance {
             type_match: compute_type_match(completion, &ty),
             exact_name_match: compute_exact_name_match(completion, &name),
-            is_local: true,
+            is_local,
             ..CompletionRelevance::default()
         });
 
         path_ref_match(completion, path_ctx, &ty, &mut item);
     };
+
+    match resolution {
+        ScopeDef::Local(local) => set_item_relevance(true, local.ty(db)),
+        ScopeDef::ModuleDef(ModuleDef::Adt(adt)) | ScopeDef::AdtSelfType(adt) => {
+            set_item_relevance(false, adt.ty(db))
+        }
+        ScopeDef::ModuleDef(ModuleDef::Function(func)) => set_item_relevance(false, func.ty(db)),
+        ScopeDef::ModuleDef(ModuleDef::Variant(variant)) => {
+            set_item_relevance(false, variant.parent_enum(db).ty(db))
+        }
+        ScopeDef::ModuleDef(ModuleDef::Const(konst)) => set_item_relevance(false, konst.ty(db)),
+        ScopeDef::ModuleDef(ModuleDef::Static(stat)) => set_item_relevance(false, stat.ty(db)),
+        ScopeDef::ModuleDef(ModuleDef::BuiltinType(bt)) => set_item_relevance(false, bt.ty(db)),
+        ScopeDef::ImplSelfType(imp) => set_item_relevance(true, imp.self_ty(db)),
+        ScopeDef::GenericParam(_)
```

Comment:
I have made the  ignored cases explicit so we get an error if new cases arise. Not sure if this is the best option I just didn't like the idea of new cases being silently ignored.